code/game/machinery/spaceheater.dm

```diff
@@ -236,6 +237,16 @@
 				cell = null
 				. = TRUE
 
+/obj/machinery/space_heater/AltClick(mob/user)
+	if(!user.canUseTopic(src, !issilicon(user)))
+		return
+	on = !on
+	mode = HEATER_MODE_STANDBY
+	user.visible_message("[user] switches [on ? "on" : "off"] \the [src].", "<span class='notice'>You switch [on ? "on" : "off"] \the [src].</span>")
+	update_icon()
+	if (on)
+		START_PROCESSING(SSmachines, src)
```

Comment:
Do not duplicate code, move this to its own proc and call it from the topic and the alt click